internal/tiltfile/tiltfile_state.go

```diff
@@ -1199,6 +1207,34 @@ func (s *tiltfileState) validateLiveUpdate(iTarget model.ImageTarget, g model.Ta
 	return nil
 }
 
+func (s *tiltfileState) validateDockerComposeVersion() error {
+	const minimumDockerComposeVersion = "v1.28.3"
+
+	dcVersion, err := s.dcCli.Version(s.ctx)
+	if err != nil {
+		logger.Get(s.ctx).Debugf("Failed to determine Docker Compose version: %v", err)
+	} else if semver.Compare(dcVersion, minimumDockerComposeVersion) == -1 {
+		return fmt.Errorf(
+			"Tilt requires Docker Compose %s+ (you have %s). Please upgrade and re-launch Tilt.",
+			minimumDockerComposeVersion,
+			dcVersion)
+	} else if semver.Major(dcVersion) == "v2" {
+		var downgradeInstructions string
+		if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+			// `docker-compose` on Docker Desktop is a shim that conditionally execs either v1 or v2
+			downgradeInstructions = "Run `docker-compose disable-v2` and re-launch Tilt."
+		} else {
+			// Compose v2 on Linux is a Docker plugin, used by running "docker compose" (notice the lack of hyphen -
+			// it's a subcommand found by searching `~/.docker/cli-plugins`), so if `docker-compose` points to v2,
+			// there must be a custom symlink; this will likely become more common over time as users want to
+			// maintain compatibility with existing tooling
+			downgradeInstructions = "Ensure `docker-compose` in your PATH points to Compose v1 and re-launch Tilt.\n"
+		}
+		logger.Get(s.ctx).Warnf("Tilt has known issues with Docker Compose v2.\n%s", downgradeInstructions)
```

Comment:
I feel like logging this as a warning makes this feel optional, but the wording of the downgrade instructions feels pretty mandatory, which is confusing. Maybe precede them with something like "If you're not in an experimental mood, ..."?

Do you think it's worth filing / linking to a GH issue to track dc v2 bugs, both so users can see known issues and we can collect new issues from users who decide to stick to v2? Or is dc v2 at a point where we don't even want to spend time on it?

Reply:
Talked offline, and message adjusted per conversation. There's also a new fallback that's meant to be long-lived for _any_ pre-release DC version to warn that you're using a beta, mostly to catch things early if Docker Desktop continues to canary RCs. Done in ab2c5b662